core/runtime.rs

```diff
@@ -2213,8 +2261,72 @@ pub mod tests {
     )
     .unwrap();
 
-    js_check(runtime.mod_evaluate(module_id));
+    runtime.mod_evaluate(module_id).unwrap();
 
     let _snapshot = runtime.snapshot();
   }
+
+  #[test]
+  fn test_error_stack() {
+    let mut runtime = JsRuntime::new(RuntimeOptions::default());
+    let result = runtime.execute(
+      "error_stack.js",
+      r#"
+function assert(cond) {
+  if (!cond) {
+    throw Error("assert");
+  }
+}
+
+function main() {
+  assert(false);
+}
+
+main();
+        "#,
+    );
+    let expected_error = r#"Error: assert
+    at assert (error_stack.js:4:11)
+    at main (error_stack.js:9:3)
+    at error_stack.js:12:1"#;
+    assert_eq!(result.unwrap_err().to_string(), expected_error);
```

Comment:
Default stack provided by V8

Reply:
We'll eventually have to opt out of this default to address this FIXME: https://github.com/denoland/deno/pull/7629/files#diff-821652b968adf3399edbd3a6dbf83cb5R195 because we'll have to register a `prepareStackTrace()` in `core` to intercept the structured stack frames without `cli`'s help.

Reply:
@nayeemrmn why is that? Tests in `core/` do not register callback for stack trace - we're using default value of `stack` property provided by V8 as string.

Reply:
> So `core/` doesn't use structured data at all

Oh, so the meaning of that FIXME isn't to make the below code work in `core`, but to move that code to `cli`. If we're okay with `core` not having access to structured frame data, which it doesn't seem to need, that would indeed be better.

Reply:
> If we're okay with core not having access to structured frame data, which it doesn't seem to need, that would indeed be better.

Yeah, I think V8 provided stack is sufficient for `core/`. 

Any ideas how to move that logic to `cli/` are welcome 😬  it will most likely require some "magical" object being assigned to error, but maybe we could treat it as JSON `Value` and do the parsing in CLI